src/OpenSSL/SSL.py

```diff
@@ -2063,6 +2063,18 @@ def sock_shutdown(self, *args, **kwargs):
         """
         return self._socket.shutdown(*args, **kwargs)
 
+    def get_certificate(self):
+        """
+        Retrieve the local certificate (if any)
+
+        :return: The local certificate
+        """
+        cert = _lib.SSL_get_certificate(self._ssl)
+        if cert != _ffi.NULL:
+            _lib.X509_up_ref(cert)
```

Comment:
Unlike SSL.get_peer_certificate, we don't own the X509 pointer. We need to increase the reference count to avoid the X509 freeing the pointer on GC.